snapcraft/internal/elf.py

```diff
@@ -209,18 +209,23 @@ def _crawl_for_path(self) -> str:
 
         logger.debug("Crawling to find soname {!r}".format(self.soname))
 
-        if self._is_valid_elf(self.soname_path):
+        valid_search_paths = [p for p in self.search_paths if os.path.exists(p)]
+        in_search_paths = any(
+            [self.soname_path.startswith(p) for p in valid_search_paths]
```

Comment:
Not that it matters here, but I think in this would be a good case for a generator (lose the list brackets):
`any(self.soname_path.startswith(p) for p in valid_search_paths)`

Reply:
I am a fan of generators in general